### PR TITLE
[sc-146473] add query params for Links in IntelliJ

### PR DESCRIPTION
**Pull request by jazanne**
also updated **[user agent](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/User-Agent)** for API client to capture IDE info

example: `launchdarkly-intellij/1.0.0-idea21 IntelliJ IDEA/202021.3.1`

**query params**

source is Full IDE version
campaign is plugin version

examples:

`https://apidocs.launchdarkly.com/?utm_source=IntelliJ%20IDEA%202021.3.1&utm_medium=ide&utm_campaign=launchdarkly-intellij/1.0.0-idea21`

`https://app.ld.catamorphic.com/docs/jwhite-dev/audit?utm_source=IntelliJ%20IDEA%202021.3.1&utm_medium=ide&utm_campaign=launchdarkly-intellij/1.0.0-idea21`

### CODE REVIEW

**Comment by jazanne** [comment]
we can optionally use utm params in app urls so that the data is available in GA

**Comment by jazanne** [comment]
given that source seems to be used only for approvals, maybe we should stick to utm params? https://launchdarkly.atlassian.net/wiki/spaces/~863195871/pages/1793196379/Feature+workflows+user+analytics+events#:~:text=source,direct

File: src/main/kotlin/com/launchdarkly/intellij/Utils.kt [hunk 1]
```diff
@@ -1,12 +1,36 @@
...
 
     fun getFlagUrl(flagKey: String): String {
         val settings = LaunchDarklyApplicationConfig.getInstance().ldState
-        return "${settings.baseUri}/${settings.project}/${settings.environment}/features/$flagKey"
+        return "${settings.baseUri}/${settings.project}/${settings.environment}/features/$flagKey?${getAppQueryParams()}"
+    }
+
+    fun getUserAgent(): String {
+        return "${ApplicationInfo.getInstance().versionName}/${ApplicationInfo.getInstance().fullVersion} ${getPluginInfo()}"
+    }
```
Comment (jazanne) [inline src/main/kotlin/com/launchdarkly/intellij/Utils.kt]: maybe we should reverse these so plugin comes first?

**Comment by carmenquan** [comment]
Will we know if users are on GoLand, WebStorm, etc based on useragent?

**Comment by jazanne** [comment]
> Will we know if users are on GoLand, WebStorm, etc based on useragent?

yep

File: src/main/kotlin/com/launchdarkly/intellij/Utils.kt [hunk 2]
```diff
@@ -1,12 +1,36 @@
...
+        return ApplicationInfo.getInstance().fullApplicationName
+    }
+
+    fun getPluginInfo(): String {
+        return "launchdarkly-intellij/${PluginManagerCore.getPlugin(PluginId.getId("com.github.intheclouddan.intellijpluginld"))?.version
+            ?: "noversion"}"
+    }
+
+    fun getAppQueryParams(): String {
+        return "source=${getIDEVersion()}&pluginVersion=${getPluginInfo()}"
```
Comment (carmenquan) [inline src/main/kotlin/com/launchdarkly/intellij/Utils.kt]: I like the idea of just sticking to utm_source, I don't think we have to reinvent the wheel or overload the source property if it was just for approvals

**Review by carmenquan — APPROVED** [review]

**Pull request merged**